packages/idyll-components/src/map.js

```diff
@@ -0,0 +1,63 @@
+const React = require('react');
+const { mapChildren } = require('idyll-component-children');
+import TextContainer from './text-container';
+
+class Map extends React.Component {
+  render() {
+    const {
+      idyll,
+      hasError,
+      updateProps,
+      children,
+      value,
+      currentValue
+    } = this.props;
+
+    if (children) {
+      return mapChildren(children, child => {
+        return value.map(val => {
+          let newProps = Object.assign({}, child.props);
+          newProps = Object.keys(child.props).reduce((props, elm) => {
+            if (props[elm] === currentValue) {
+              props[elm] = val;
+              return props;
+            }
+            return props;
+          }, newProps);
+          return (
+            <TextContainer>
```

Comment:
I don't think we need this `TextContainer` here

Reply:
I have updated this to remove the TextContainer.